packages/flutter_tools/lib/src/run_hot.dart

```diff
@@ -498,8 +498,8 @@ class HotRunner extends ResidentRunner {
 
   void _resetDirtyAssets() {
     for (final FlutterDevice? device in flutterDevices) {
-      device!.devFS!.assetPathsToEvict.clear();
-      device.devFS!.shaderPathsToEvict.clear();
+      device!.devFS?.assetPathsToEvict.clear();
```

Comment:
devFS can be null if we hit this code path either before the device has been set up or after its been torn down. The latter seems more likely - perhaps if the user triggers a reload that stalls and then they exit? We hit this issue before and I was never able to repro, in that case something like the following would be more obvious for future readers.

```
final DevFS? devfs = device.devFS;
if (devfs == null) {
 // note about why this happens
  continue;
}
```

Reply:
SGTM, however I have no idea why it happens. Should I just link to the issue?

Reply:
Yeah, that is a reasonable approach. We need this null check to prevent a crash, but we don't know why exactly see issue# or somesuch